spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/core/PubSubConfiguration.java

```diff
@@ -54,34 +69,74 @@ public Health getHealth() {
     return health;
   }
 
-  public ConcurrentMap<String, Subscriber> getSubscription() {
-    return this.subscription;
+  /**
+   * This method will be called by Spring Framework when binding user properties.
+   * Also potentially useful for tests.
+   *
+   * @param map map of user-defined properties.
+   */
+  public void setSubscription(Map<String, Subscriber> map) {
+    Assert.isNull(this.fullyQualifiedSubscriptionProperties,
+        "Pub/Sub properties have already been initialized; cannot update subscription properties");
+
+    this.subscription = map;
   }
 
-  public Subscriber getSubscriber(String name, String projectId) {
-    ProjectSubscriptionName fullyQualifiedName =
-        PubSubSubscriptionUtils.toProjectSubscriptionName(name, projectId);
-    String fullyQualifiedSubscriptionKey = fullyQualifiedName.toString();
+  public Map<ProjectSubscriptionName, Subscriber> getFullyQualifiedSubscriberProperties() {
+    Assert.notNull(this.fullyQualifiedSubscriptionProperties, "Please call initialize() prior to retrieving properties.");
+    return this.fullyQualifiedSubscriptionProperties;
+  }
 
-    if (this.subscription.containsKey(fullyQualifiedSubscriptionKey)) {
-      return this.subscription.get(fullyQualifiedSubscriptionKey);
-    }
+  /**
+   * Standardizes all subscription properties to be fully qualified. Not thread-safe.
```

Comment:
It could be helpful to mention in the javadoc that the `fullyQualifiedName` property takes precedence over the generated one.